src/utils/color-helpers.js

```diff
@@ -0,0 +1,43 @@
+export const ALL_OTHERS_LABEL = "All Others"
+export const ALL_OTHERS_COLOR = "#888888"
+
+export function maybeUpdateDomainRange(
+  chart,
+  data,
+  dataAccessor,
+  domain,
+  range,
+  wrapKey = false
+) {
+  const dataArr = data.map(dataAccessor)
+  const domainRangeMap = new Map([...domain].map((key, i) => [key, range[i]]))
+  const dataMap = new Map(
+    [...dataArr].map((key, i) => [
+      key,
+      domainRangeMap.get(key) ??
+        chart.getColor(wrapKey ? { key0: key } : key, i)
+    ])
+  )
+  chart.customDomain([...dataMap.keys()])
```

Comment:
I don't think these spread operators are necessary either, `keys` and `values` return new arrays